.github/workflows/release.yml

```diff
@@ -0,0 +1,57 @@
+name: Release Images and Binaries
+
+on:
+  push:
+    tags:
+      - 'v**'
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v4
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version-file: ./go.mod
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Build images
+      run: make docker-build
+    - name: Tag and push ClusterLink images with tag 'latest'
+      run: make push-image
+    - name: Tag and push ClusterLink images latest with tag ${{ github.ref_name }}
+      run: make push-image IMAGE_VERSION=${{ github.ref_name }}
+    - name: Build and compress linux binaries
+      run: |
+        GOOS=linux GOARCH=amd64 CGO_ENABLED=0 make cli-build BIN_DIR=linux
+        tar -czvf clusterlink-linux-amd64.tar.gz --transform 's/linux/clusterlink/' ./linux
+    - name: Build and compress macOS binaries
+      run: |
+        GOOS=darwin GOARCH=amd64 CGO_ENABLED=0 make cli-build BIN_DIR=os
```

Comment:
does GH actions support `foreach`, shell like, syntax?
It may be clearer to iterate on that (or on concat of "GOOS/GOARCH" pairs split internally in the loop